website/content/docs/agentandproxy/index.mdx

```diff
@@ -0,0 +1,126 @@
+    if err != nil {
+        return "", fmt.Errorf("unable to read secret: %w", err)
+    }
+
+    data := secret.Data["data"].(map[string]interface{})
+
+    ...snip...
+}
+```
+
+For some Vault deployments, making (and maintaining) these changes to
+applications may not be a problem, and may actually be preferred. This may be
+applied to scenarios where you have a small number of applications, or you want
+to keep strict, customized control over how each application interacts with
+Vault. However, in other situations where you have a large number of
+applications, as in large enterprises, you may not have the resources or expertise
+to update and maintain the Vault integration code for every application. When
+third party applications are being deployed by the application, it is prohibited
+to add the Vault integration code.
+
+Vault Agent and Vault Proxy aim to remove this initial hurdle to adopt Vault by providing a
+more scalable and simpler way for applications to integrate with Vault. Vault Agent can
+obtain secrets and provide them to applications, and Vault Proxy can act as
+a proxy between Vault and the application, optionally simplifying the authentication process
+and caching requests.
+
+## What is Vault Agent?
+
+[Vault Agent][vaultagent] is a client daemon that provides the following features:
+
+- [Auto-Auth][autoauth] - Automatically authenticate to Vault and manage the
+  token renewal process for locally-retrieved dynamic secrets.
+- [Caching][caching] - Allows client-side caching of responses containing newly
+  created tokens and responses containing leased secrets generated off of these
+  newly created tokens. The agent also manages the renewals of the cached tokens and leases.
+- [Windows Service][winsvc] - Allows running the Vault Agent as a Windows
+  service.
+- [Templating][template] - Allows rendering of user-supplied templates by Vault
+  Agent, using the token generated by the Auto-Auth step.
+
+## What is Vault Proxy?
+
+[Vault Proxy][vaultproxy] is a client daemon that provides the following features:
+
+- [Auto-Auth][autoauth] - Automatically authenticate to Vault and manage the
+token renewal process for locally-retrieved dynamic secrets.
+- [API Proxy][apiproxy] - Allows Vault Proxy to act as a proxy for Vault's API,
+optionally using (or forcing the use of) the Auto-Auth token, allowing downstream
+applications to use its APIs as if they were Vault's.
+- [Caching][caching] - Allows client-side caching of responses containing newly
+created tokens and responses containing leased secrets generated off of these
+newly created tokens. The agent also manages the renewals of the cached tokens and leases.
+
+## Auto-Auth
+
+[Vault Agent][vaultagent] and [Vault Proxy][vaultproxy] allow for easy authentication to Vault
+in a wide variety of environments. Please see the [Auto-Auth docs][autoauth] for information.
+
+Auto-Auth functionality takes place within an `auto_auth` configuration stanza.
```

Comment:
Is it worth providing an example here, or linking to an example of the configuration stanza?